mne/io/meas_info.py

```diff
@@ -1029,6 +1029,11 @@ def pick_channels(self, ch_names, ordered=False):
 
         .. versionadded:: 0.20.0
         """
+        warn('This method is deprecated and will be removed after 1.1. To '
+             'pick channels from an instance, use '
+             'inst.pick_channels(ch_names). To pick channels from an Info, '
+             'use mne.pick_info(info, sel).', DeprecationWarning)
```

Comment:
Instead of doing this, use the `@deprecated` decorator like:
```
@deprecated('info.pick_channels is deprecated, use inst.pick_channels instead.')
def pick_channels(...):
```
I wouldn't mention pick_info in this runtime-emitted DeprecationWarning because it's rare/expert mode, and the risk of a more basic user seeing `pick_info` as an option and just dropping that in is too high, I think. We can keep the `pick_info` note in the `latest.inc`, though, as a guide for the experts.